test/sqllogictest/temporal.slt

```diff
@@ -0,0 +1,45 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
```

Comment:
Those two definitions appear to be valid and yet they are rejected:

```
# mz_logical_timestamp is on the right side and not the left
materialize=> create view v1 as select * from temporal where tm1 < mz_logical_timestamp();
ERROR:  "MzLogicalTimestamp in unsupported position"
```

and

```#nested condition with mz_logical_timestamp

materialize=> create view v1 as select * from temporal where mz_logical_timestamp() > tm1 or mz_logical_timestamp() < tm2;
ERROR:  "MzLogicalTimestamp in unsupported position"
```

Reply:
Let's see. The first one is supposed to work (I'll check it out). The second one has an `or` in there, which is meant to be ruled out (though perhaps it is not as clearly stated as it could be: the docs say 
> This mechanism allows us to support `WHERE` clauses in which `mz_logical_timestamp()` is directly related to an expression not containing `mz_logical_timestamp()`, or conjunctions (`AND`) of these clauses.

but I could imagine we need to be more clear or explain better.

Reply:
The first case does indeed seem to be an error in how I was testing for the "order switching" logic. Fix incoming!

Reply:
I think 9c81b7e should fix the re-ordering example. No rush, but lmk if you get a chance to re-test that (it loaded for me, but I didn't e.g. check answers or anything).

Reply:
Yes your fix seems to do the trick, thanks. I now understand why disjunctions do not work, but it would be nice if the documentation and the error message should be made more clear.